packages/contracts/deploy/helpers.ts

```diff
@@ -167,26 +176,71 @@ export async function checkSetManagingDao(
   }
 }
 
-export async function checkPermission(
-  permissionManager: Contract,
-  where: string,
-  who: string,
-  permission: string,
-  data = '0x'
-) {
+export interface CheckPermission {
+  isGrant: boolean;
+  permissionManager: ethers.Contract;
+  where: string;
+  who: string;
+  permission: string;
+  data?: string;
+}
+
+export async function checkPermission({
+  isGrant,
+  permissionManager,
+  where,
+  who,
+  permission,
+  data = '0x',
+}: CheckPermission) {
   const permissionId = ethers.utils.id(permission);
   const isGranted = await permissionManager.callStatic.isGranted(
     where,
     who,
     permissionId,
     data
   );
-  if (!isGranted) {
+  if (!isGranted && isGrant) {
     throw new Error(
       `${who} doesn't have ${permission} on ${where} in ${permissionManager.address}`
     );
+  } else if (isGranted && !isGrant) {
+    throw new Error(
+      `${who} have ${permission} on ${where} in ${permissionManager.address}`
+    );
   }
 }
 
+export interface ManagePermission {
```

Comment:
do we really need 2 `interfaces` ? One is enough. Maybe to use both, you could have such a type.

```
export interface CheckPermission {
  isGrant: boolean;
  permissionManager: ethers.Contract;
  where: string | {name: string; address: string};
  who: string | {name: string; address: string};
  permission: string;
  data?: string;
}
```

and I'd use `type` keyword for it !

Reply:
amended